components/org.wso2.carbon.identity.auth.valve/src/test/java/org/wso2/carbon/identity/auth/valve/internal/AuthenticationValveServiceComponentTest.java

```diff
@@ -0,0 +1,84 @@
+
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.osgi.service.component.ComponentContext;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.auth.service.AuthenticationManager;
+import org.wso2.carbon.identity.auth.service.factory.AuthenticationRequestBuilderFactory;
+
+import java.util.List;
+
+
+public class AuthenticationValveServiceComponentTest extends PowerMockTestCase {
+
+    @Mock
+    private ComponentContext componentContext;
+    @Mock
+    private AuthenticationValveServiceHolder authenticationValveServiceHolder;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testActivate() throws Exception {
+        AuthenticationValveServiceComponent authenticationValveServiceComponent = new
+                AuthenticationValveServiceComponent();
+        authenticationValveServiceComponent.activate(componentContext);
+    }
+
+    @Test
+    public void testAddAuthenticationRequestBuilderFactory() throws Exception {
+        AuthenticationRequestBuilderFactory authenticationRequestBuilderFactory = AuthenticationRequestBuilderFactory
+                .getInstance();
+        AuthenticationValveServiceComponent authenticationValveServiceComponent = new
+                AuthenticationValveServiceComponent();
+        authenticationValveServiceComponent.addAuthenticationRequestBuilderFactory(authenticationRequestBuilderFactory);
+
+        AuthenticationValveServiceHolder authenticationValveServiceHolder = AuthenticationValveServiceHolder
+                .getInstance();
+
+        List<AuthenticationRequestBuilderFactory> requestBuilderFactories = authenticationValveServiceHolder
+                .getRequestBuilderFactories();
+        for (AuthenticationRequestBuilderFactory builderFactory : requestBuilderFactories) {
+            Assert.assertEquals(authenticationRequestBuilderFactory, builderFactory);
+        }
+    }
+
+    @Test
+    public void testSetAuthenticationManager() throws Exception {
+        AuthenticationManager authenticationManager = AuthenticationManager.getInstance();
+        AuthenticationValveServiceComponent authenticationValveServiceComponent = new
+                AuthenticationValveServiceComponent();
+        authenticationValveServiceComponent.setAuthenticationManager(authenticationManager);
+
+        AuthenticationValveServiceHolder authenticationValveServiceHolder = AuthenticationValveServiceHolder
+                .getInstance();
+
+        List<AuthenticationManager> authenticationManagers = authenticationValveServiceHolder
+                .getAuthenticationManagers();
+        for (AuthenticationManager manager : authenticationManagers) {
+            Assert.assertEquals(authenticationManager, manager);
+        }
```

Comment:
Better to assert when list is empty as well

Reply:
Done